src/components/Text/Text.styles.ts

```diff
@@ -0,0 +1,49 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { TextProps, TextSize } from './Text';
+import { ellipsisStyle, fontWeightBold, fontSizes, lineHeights, textColors, Sizes } from '../../util/styles/fonts';
+import { IRawStyle } from '@uifabric/styling';
+
+const heights: Sizes = {
+  xSmall: '1.6rem',
+  small: '1.6rem',
+  mediumSub: '2rem',
+  medium: '2rem',
+  large: '2.4rem',
+  xLarge: '2.8rem',
+  xxLarge: '2.8rem',
+};
+
+const getHeight = (size?: TextSize) => {
+  return size ? heights[size] : undefined;
+};
+
+const verticalAligns: Sizes = {
+  xSmall: '-0.4rem',
+  small: '-0.3rem',
+  mediumSub: '-0.5rem',
+  medium: '-0.4rem',
+  large: '-0.5rem',
+  xLarge: '-0.5rem',
+  xxLarge: '-0.4rem',
+};
+
+const getVerticalAlign = (size?: TextSize) => {
+  return size ? verticalAligns[size] : '-0.4rem';
```

Comment:
this should fall back to undefined, unless maxWidth was provided then 0.4rem

Reply:
That is the logic, but it's split between this and line 47.  I've renamed this to `getVerticalAlignForMaxWidth` to make it clearer.